base/compiler/ssair/inlining.jl

```diff
@@ -597,18 +608,40 @@ function ir_inline_unionsplit!(compact::IncrementalCompact, idx::Int, argexprs::
     end
     bb += 1
     # We're now in the fall through block, decide what to do
-    if !fully_covered
+    if !handled_all_cases
         ssa = insert_node_here!(compact, NewInstruction(stmt, typ, line))
         push!(pn.edges, bb)
         push!(pn.values, ssa)
         insert_node_here!(compact, NewInstruction(GotoNode(join_bb), Any, line))
         finish_current_bb!(compact, 0)
+    elseif !fully_covered
+        ir_inline_method_error!(compact, idx, argexprs, interp)
+        finish_current_bb!(compact, 0)
+        ncases == 0 && return insert_node_here!(compact, NewInstruction(nothing, Any, line))
     end
-
     # We're now in the join block.
     return insert_node_here!(compact, NewInstruction(pn, typ, line))
 end
 
+"""
+    ir_inline_method_error!
+
+Insert a static method error for the call expression at `idx` in `compact`
+The call is inserted at the current IncrementalCompact cursor
+"""
+function ir_inline_method_error!(compact::IncrementalCompact, idx::Int, argexprs::Vector{Any}, interp::AbstractInterpreter)
+    line = compact.result[idx][:line]
+    world_age_stmt = Expr(:foreigncall, :(:jl_get_tls_world_age), UInt, svec(), 0, :(:ccall))
+    flags = recompute_effects_flags(optimizer_lattice(interp), world_age_stmt, UInt, compact)
+    world_age = insert_node_here!(compact, NewInstruction(world_age_stmt, UInt, NoCallInfo(), line, flags))
+    argtuple = insert_node_here!(compact, NewInstruction(Expr(:call, tuple, argexprs[2:end]...), Tuple, line))
```

Comment:
```suggestion
    argtuple = insert_node_here!(compact, NewInstruction(Expr(:call, GlobalRef(Core, :tuple), argexprs[2:end]...), Tuple, line))
    world_age = insert_node_here!(compact, NewInstruction(world_age_stmt, UInt, NoCallInfo(), line, flags))
```

Reply:
Question out of personal ignorance: is there any particular reason for the swap?